Extension/src/LanguageServer/configurations.ts

```diff
@@ -762,8 +766,14 @@ export class CppProperties {
         }
         paths = this.resolveDefaults(paths, defaultValue);
         paths.forEach(entry => {
-            const entries: string[] = util.resolveVariables(entry, env).split(path.delimiter).map(e => glob ? this.resolvePath(e, false) : e).filter(e => e);
-            resolvedVariables.push(...entries);
+            const resolvedVariable: string = util.resolveVariables(entry, env);
```

Comment:
I was thinking we could just resolve undefined environment variables to the empty string in `util.resolveVariables`.  Then you wouldn't have to do any checks about leftover `${env:}` stuff.

Reply:
I believe most terminals behave that way when resolving env variables that don't exist (resolve to empty string)

Reply:
I purposely left the unresolved variable as is so that it will show up as unresolved in the logs.

Reply:
I'll make the `util.resolveVariables` return empty if an env is unresolved and have the caller save the original variable string.

Reply:
After looking at the other callers and scenarios that use the utilities `resolveVariables`, I determined that returning an empty string for an unresolved `env` variable would break other case or scenarios where an empty string is valid such as `""` value for compilerPath or where `""` is returned for `${default}`.

Rather than changing the expected behavior of this utility function and changing the callers and potentially breaking them, it is less risky to have the caller check the returned value for further post processing. I removed the helper function to check if a returned value has "env" or not, as that currently is only needed in two places.

If needed, I propose a different PR to change the design/usage of all utilities that resolve variables to ensure a consistent  /robust behavior for all callers.